src/Vimeo/ABLincoln/Operators/Random/WeightedChoice.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Vimeo\ABLincoln\Operators\Random;
+
+use \Vimeo\ABLincoln\Operators\RandomOperator;
+
+/**
+ * Select an element from a choices array according to given probabilities
+ */
+class WeightedChoice extends RandomOperator
+{
+    /**
+     * The operator requires an set of choices to draw from and weights to use
+     *
+     * @return array the array of required parameters
+     */
+    public function options()
+    {
+        return array(
+            'choices' => array(
+                'required' => 1,
+                'description' => 'elements to draw from'
+            ),
+            'weights' => array(
+                'required' => 1,
+                'description' => 'probability of draw'
+            )
+        );
+    }
+
+    /**
+     * Choose an element with weighted probability from the parameter array
+     *
+     * @return mixed the element chosen from the given array
+     */
+    protected function _simpleExecute()
+    {
+        $choices = $this->parameters['choices'];
+        $weights = $this->parameters['weights'];
+        if (empty($choices)) {
+            return array();
+        }
+        $cum_weights = array_combine($choices, $weights);
```

Comment:
Hotfix in f9692bd: added some tests and used `array_key_values` in weightedChoice to allow flexibility for users to supply weights either as a vector `array(1, 2)` or as an associative array `array(choice1 => 1, choice2 => 2)`